XLS-30/README.md

```diff
@@ -0,0 +1,1254 @@
+| `lsfAMM`      | 0x40000000  |   If set, it indicates that the object represents an AMM instance|
+
+Additionally, we introduce a new field called `AMMID` to the `AccountRoot` object. It replaces directory entry to link account root and AMM object. 
+
+| Field Name        |     Required?      | JSON Type | Internal Type |
+| ----------------- | :----------------: | :-------: | :-----------: |
+| `AMMID` | :heavy_check_mark: | `string`  |   `UINT256`    |
+
+`AMMID` specifies the `AMMID` as described below.
+
+The unique ID of the **`AccountRoot`** object, a.k.a. **`AccountRootID`** is computed as follows:
+- for (i = 0; i <= 256; i--)
+  - Compute `AccountRootID` = `SHA512-Half`(i || [Parent Ledger Hash](https://xrpl.org/ledgerhashes.html) || `AMMID`)
+  - If the computed `AccountRootID` exists, repeat
+    - else, return `AccountRootID`
+ 
+#### **`AMM`** ledger entry 
+The unique ID of the new **`AMM`** object , a.k.a **`AMMID`** is computed as follows:
+- Calculate the `SHA512-Half` of some of the following values:
+  - The `issuer` of the issued asset;
+  - The `currency` of the issued asset;
+  - The `issuer` of the second issued asset, if there exists one;
+  - The `currency` of the second issued asset, if there exists one;
+  <!-- `XRP` if the `Balance` field exists, i.e. if one of the assets is XRP "GT: why is this needed? We just use issuer/currency of each asset. XRP defaults both to 0."-->
+  
+The order of the fields to compute the hash is decided canonically. The **`AMMID`** associated with this **`AccountRootID`** is this hash to ensure the uniqueness of the AMM instance. The applications can look-up the **`AccountRootID`** for a specific AMM instance by providing the asset pair for that instance.
+
+The **`AMM`** ledger entry contains the following fields for the **`AccountRoot`** object that represents the AMM instance.
+
+1. **`AccountID`** specifies the ID of the `AccountRoot` object associated with this `AMM` ledger entry.
+2. **`TradingFee`** specifies the fee, in basis point, to be charged to the traders for the trades executed against this AMM instance. Valid values for this field are between 0 and 1000 inclusive. A value of 1 is equivalent to 1/10 bps or 0.001%, allowing trading fee between 0% and 1%. Trading fee is a percentage of the trade size. It is charged on the asset being deposited for the `AMMDeposit` (if applicable) and `Payment` transactions and on the asset being withdrawan for the `AMMWithdraw` (if applicable) transaction. This fee is added to the AMM instance's pools and is distributed to the LPs in proportion to the `LPTokens` upon liquidity removal.
+
+3. **`VoteSlots`** represents an array of `Vote` objects.
+4. **`AuctionSlot`** represents the `Auction` object.
+5. **`LPTokenBalance`** specifies the balance of outstanding liquidity Provider Tokens (LPTokens).
+6. **`Asset`** specifies the one of the assets of the AMM instance. 
+7. **`Asset2`** specifies the other asset of the AMM instance.
+   
```

Comment:
There is a new field `OwnerNode`. It's a `page` hint for the `DirectoryNode` entry to link the account root and AMM object.